devito/finite_differences/derivative.py

```diff
@@ -222,25 +220,37 @@ def _process_weights(cls, **kwargs):
 
     def __call__(self, x0=None, fd_order=None, side=None, method=None, weights=None):
         side = side or self._side
+        method = method or self._method
+        weights = weights if weights is not None else self._weights
+
         x0 = self._process_x0(self.dims, x0=x0)
         _x0 = frozendict({**self.x0, **x0})
-        if self.ndims == 1:
-            fd_order = fd_order or self._fd_order
-            method = method or self._method
-            weights = weights if weights is not None else self._weights
-            return self._rebuild(fd_order=fd_order, side=side, x0=_x0, method=method,
-                                 weights=weights)
-
-        # Cross derivative
+
         _fd_order = dict(self.fd_order.getters)
         try:
             _fd_order.update(fd_order or {})
-            _fd_order = tuple(_fd_order.values())
-            _fd_order = DimensionTuple(*_fd_order, getters=self.dims)
+        except TypeError:
+            assert self.ndims == 1
+            _fd_order.update({self.dims[0]: fd_order or self.fd_order[0]})
         except AttributeError:
-            raise TypeError("Multi-dimensional Derivative, input expected as a dict")
+            raise TypeError("fd_order incompatible with dimensions")
+
+        # In case this was called on a cross derivative we need to propagate
+        # the call to the nested derivative
+        if isinstance(self.expr, Derivative):
+            _fd_orders = {k: v for k, v in _fd_order.items() if k in self.expr.dims}
+            _x0s = {k: v for k, v in _x0.items() if k in self.expr.dims and
```

Comment:
>Should this consolidate them to u.dx2(x0=x)

Definitely not that'd be terrible. This will only propagate x0 to the inner derivative for dimensions not for the current one. so `u.dx(x0=x+hx/2).dx(x0=x-hx/2)` the `x-hx/2` is the outer one but since `x` is the outer derivative dim this x0 is not passed through, and `x+hx/2` is just for the inner dim

Reply:
Seems reasonable